### PR TITLE
Create factorial_iterative.py

### PR DESCRIPTION
**Pull request by pooja-git11**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/blob/master/Travis_CI_tests_are_failing.md

File: maths/factorial_iterative.py [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 
 n=int(input("Enter a number"))
-
 def factorial(n):
     fact=1
```
Comment (cclauss) [inline maths/factorial_iterative.py]: The function should raise an ValueError just like __math.factorial()__ does:

% __python3__
```
>>> import math
>>> math.factorial(-1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: factorial() not defined for negative values
```

**Comment by pooja-git11** [comment]
Respceted cclauss,
What's the problem in my code ?i am not able to get your point
if possible can you help me to find out?.

File: maths/factorial_iterative.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# factorial of a number
+
+
+n=int(input("Enter a number"))
```
Comment (cclauss) [inline maths/factorial_iterative.py]: This line will fail our automated testing unless it is indented under `if __name__ == "__main__":` as discussed in CONTRIBUTING.md

File: maths/factorial_iterative.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+def factorial(n):
```
Comment (cclauss) [inline maths/factorial_iterative.py]: See other functions in this repo to see how type hints and doctests work.  They are required.

**Pull request merged**